enhancements/rhcos/multi-stream.md

```diff
@@ -0,0 +1,497 @@
+---
+title: CoreOS Multi Stream
+authors:
+  - @sdodson
+  - @jlebon
+  - @yuqi-zhang
+reviewers: 
+  - @jlebon        # RHCOS build, shipping, testing
+  - @patrickdillon # Installer boot image selection and container native image refs
+  - @yuqi-zhang    # MCO items and Boot Image management items
+  - @cewong        # NodePool stream selection, boot image selection, and container native image refs
+  - @joepvd      # Delivering multiple RHCOS streams in a single release
+  - @peterhunt     # Node team concerns about supporting two major OSes
+approvers:
+  - @mrunalp  
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2024-08-21
+last-updated: 2024-08-21
+tracking-link: 
+  - https://issues.redhat.com/browse/OCPSTRAT-1150
+
+# CoreOS Multi Stream
+
+## Summary
+
+Starting as early as OpenShift Container Platform (OCP) 4.19 we will ship
+and concurrently support two CoreOS streams, each based on one of the two
+most recent major RHEL versions. All components responsible for lifecycling
+hosts and host configuration will be extended to select from multiple streams
+and configure their resources appropriately.
+
+Installer, Machine Config Operator, RHCOS builds, Hypershift, and at least ART
+delivery pipelines will need to coordinate to deliver this work.
+
+
+## Motivation
+
+RHEL Lifecycles and the hardware certified or supported by a given RHEL version
+are substantially longer than OCP lifecycles, therefore in order to balance
+support of latest hardware and hardware still supported through end of RHEL
+Maintenance Phases we must support at least two major OS versions.
+
+### User Stories
+
+* As an OpenShift admin in an environment where hardware certification is not
+a concern, I don't want to think about OS major version selection so that I'm
+using the OS stream that Red Hat chooses.
+
+* As an OpenShift admin, with hardware certified or supported only by a
+specific OS stream I want to specify the OS Stream for that hardware grouped
+by installer pools 'ControlPlane' and 'Compute'.
+
+* As an OpenShift admin adding newer hardware to an existing cluster I want the
+new hardware to boot, run, and update from a specific OS stream.
+
+* As an OpenShift admin wishing to migrate existing hosts to a newer stream I
```

Comment:
We debated this live, but I want to write things down here for consistency. I have no opposition to making it obvious/easy for admins to use CAPI to spawn separate RHEL-$next hosts for testing, etc.

But I am pretty confident in stating that we can support seamless in-place upgrades from 9 to 10 for the majority of e.g. cloud-deployed clusters. Take our own Prow clusters for example. I bet we can just flip the flag to run those on rhel10 and watch it roll out in place by default and it would Just Work.